.github/workflows/java-integration-tests.yml

```diff
@@ -1,30 +1,68 @@
 name: Run Java Integration Tests and Samples
 
 on:
+  workflow_dispatch:
   push:
     branches: [ "experimental-java", "java-dev" ]
+    paths:
+      - 'java/**'
+      - 'samples/java/**'
   pull_request:
     branches: [ "experimental-java", "java-dev" ]
+    paths:
+      - 'java/**'
+      - 'samples/java/**'
 
-jobs:
-  build:
+permissions:
+  contents: read
 
+jobs:
+  java-integration-tests:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        java-versions: [8, 11, 17]
+
+    name: Java Tests on JDK${{ matrix.java-versions }}
 
     steps:
-      - uses: actions/checkout@v3
-      - name: Set up JDK 17
-        uses: actions/setup-java@v3
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      # Need to use JDK 11 to build for JDK 8
+      - name: Set JDK
```

Comment:
Where the steps are the same between integration-tests.yml and java-build.yml, maybe the step should be in its own yml? For instance create a set-sdk.yml and use `- template: set-sdk.yml`

Reply:
There is no `template` property on Github but there is something called `Composite Action` that has a similar behaviour but it will require some investigation. What do you think?